integration_tests/variable_decl_01.py

```diff
@@ -0,0 +1,19 @@
+from ltypes import i32, i64
+from numpy import empty, int64
+
+def f(n: i32, m: i32):
+    l: i32 = 2
+    a: i64[n, m, l] = empty((n, m, l), dtype=int64)
```

Comment:
`a` depends on `n` and `m`, which is fine, every time the function gets called with some arguments, it will allocate a "stack" (or equivalent) array of length that is an expression of `n` and `m`.

What is `l`? I thought we generate an assignment for it like `l = 2` , but by looking at the ASR using this PR:
```
                            a:
                                (Variable 
                                    171 
                                    a 
                                    [n
                                    m
                                    l] 
                                    Local 
                                    () 
                                    () 
                                    Default 
                                    (Integer 8 [((IntegerConstant 0 (Integer 4 []))                                 (Var 171 n))
                                    ((IntegerConstant 0 (Integer 4 [])) 
                                    (Var 171 m))
                                    ((IntegerConstant 0 (Integer 4 [])) 
                                    (Var 171 l))]) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
                            l:
                                (Variable 
                                    171 
                                    l 
                                    [] 
                                    Local 
                                    (IntegerConstant 2 (Integer 4 [])) 
                                    () 
                                    Default 
                                    (Integer 4 []) 
                                    Source 
                                    Public 
                                    Required 
                                    .false.
                                ), 
```
It shows that `l` is actually a constant. In that case, we can use it in declarations and yes, doing the dependency is fine.

So I guess this is fine.